src/core/RandomVariables.jl

```diff
@@ -170,6 +380,27 @@ function link!(vi::UntypedVarInfo, spl::Sampler)
         @warn("[Turing] attempt to link a linked vi")
     end
 end
+function link!(vi::TypedVarInfo, spl::Sampler)
+    vns = getvns(vi, spl)
+    space = getspace(spl)
+    return _link!(vi.vis, vi, vns, space)
+end
+@inline function _link!(vis::NamedTuple{names}, vi, vns, space) where {names}
+    length(names) === 0 && return nothing
+    f = names[1]
```

Comment:
A bit confused by the variables here, what is `f`? Also we are calling `getfield(vns, f)` multiple times later, can we assign it to some local variable with a name?

Reply:
`f` is the first variable's symbol.

Reply:
But using recursion, this "first" variable will be a different variable in each level of the recursion.

Reply:
But for the loop below we are calling `getfield(vns, f)` with same `vns` and `f` all the time do we?

Reply:
Yes, so we are iterating over all the `vn`s of the first symbol, then all the `vn`s of the second, and so on. Each symbol has a `vns` vector because it can consist of multiple random variables, e.g. multiple univariate variables arranged in vector form or multiple multivariate variables arranged in matrix form, etc. The last line is where recursion happens.